src/main/java/com/tomtom/services/configuration/implementation/Configuration.java

```diff
@@ -658,50 +694,69 @@ private static boolean checkNodeMatchStringsChildren(@Nullable final List<NodeDT
      * @param tree     Node to expand.
      * @param included Memory of which include files were processed.
      * @throws IncorrectConfigurationException If include recursion was detected.
+     * @return Replacements for the node that was just expanded.
      */
-    private static void expandAllIncludes(
+    private static List<NodeDTO> expandAllIncludes(
             @Nonnull final NodeDTO tree,
             @Nonnull final List<String> included) throws IncorrectConfigurationException {
         final String include = tree.getInclude();
-        if (include != null) {
+        final String includeArray = tree.getIncludeArray();
+        if (include != null || includeArray != null) {
```

Comment:
And perhaps it's more readable to bail out early if either is null, rather than indent the entire method in the if.